### DISCUSSION TITLE
no variable found for substitution expression Error

### DISCUSSION THREAD

**Discussion by agniksr** [idaholab/moose#27917] [Q&A General]
### Check these boxes if you have followed the posting rules.

- [X] Q&A General is the most appropriate section for my question
- [X] I have consulted the posting Guidelines on the Discussions front page
- [X] I have searched the Discussions forum and my question has not been asked before
- [X] I have searched the MOOSE website and the documentation does not answer my question
- [X] I have formatted my post following the posting guidelines (screenshots as a last resort, triple back quotes around pasted text)

### Question

The input file in the examples of HeatConductionMaterial named : (modules/heat_transfer/test/tests/gap_heat_transfer_mortar/large_gap_heat_transfer_test_rz_cylinder.i)

displays the following error when i run it:
```
*** ERROR ***
/home/agniksr/projects/leo/Input Files/peacock_run_mesh_tmp.i:7.5: no variable 'core_outer_radius' found for substitution expression
/home/agniksr/projects/leo/Input Files/peacock_run_mesh_tmp.i:7.5: no variable 'rpv_core_gap_size' found for substitution expression
/home/agniksr/projects/leo/Input Files/peacock_run_mesh_tmp.i:7.5: no variable 'rpv_width' found for substitution expression
/home/agniksr/projects/leo/Input Files/peacock_run_mesh_tmp.i:16.5: no variable 'core_outer_radius' found for substitution expression
/home/agniksr/projects/leo/Input Files/peacock_run_mesh_tmp.i:30.5: no variable 'rpv_inner_radius' found for substitution expression
/home/agniksr/projects/leo/Input Files/peacock_run_mesh_tmp.i:31.5: no variable 'rpv_outer_radius' found for substitution expression
/home/agniksr/projects/leo/Input Files/peacock_run_mesh_tmp.i:63.5: no variable 'core_blocks' found for substitution expression
/home/agniksr/projects/leo/Input Files/peacock_run_mesh_tmp.i:78.5: no variable 'rpv_outer_htc' found for substitution expression
/home/agniksr/projects/leo/Input Files/peacock_run_mesh_tmp.i:79.5: no variable 'rpv_outer_Tinf' found for substitution expression
/home/agniksr/projects/leo/Input Files/peacock_run_mesh_tmp.i:100.5: no variable 'core_blocks' found for substitution expression
/home/agniksr/projects/leo/Input Files/peacock_run_mesh_tmp.i:115.5: no variable 'core_blocks' found for substitution expression
/home/agniksr/projects/leo/Input Files/peacock_run_mesh_tmp.i:121.5: no variable 'core_blocks' found for substitution expression
/home/agniksr/projects/leo/Input Files/peacock_run_mesh_tmp.i:127.5: no variable 'core_blocks' found for substitution expression
/home/agniksr/projects/leo/Input Files/peacock_run_mesh_tmp.i:132.5: no variable 'rpv_blocks' found for substitution expression
/home/agniksr/projects/leo/Input Files/peacock_run_mesh_tmp.i:138.5: no variable 'rpv_blocks' found for substitution expression
/home/agniksr/projects/leo/Input Files/peacock_run_mesh_tmp.i:144.5: no variable 'rpv_blocks' found for substitution expression
/home/agniksr/projects/leo/Input Files/peacock_run_mesh_tmp.i:149.5: no variable 'core_blocks' found for substitution expression
/home/agniksr/projects/leo/Input Files/peacock_run_mesh_tmp.i:155.5: no variable 'rpv_outer_Tinf' found for substitution expression
/home/agniksr/projects/leo/Input Files/peacock_run_mesh_tmp.i:156.5: no variable 'rpv_outer_htc' found for substitution expression
```

Could someone please help with this

**Comment by GiudGiud** [comment]
Hello

This input `large_gap_heat_transfer_test_rz_cylinder.i` runs fine for me.
You seem to be running a different input: `/peacock_run_mesh_tmp.i`
where did you find this?

Guillaume

**Comment by agniksr** [comment]
Yeah i just copied that input file into my own app just to see if it runs. It has the HeatTransfer module enabled

**Reply by GiudGiud** [reply]
how are you running this input?

**Reply by agniksr** [reply]
i am using peacock -i test.i

**Reply by GiudGiud** [reply]
Just use peacock, without an input file, then open the other input

**Reply by agniksr** [reply]
Still gives me the same error

'''
Running command: /home/agniksr/projects/leo/leo-opt Outputs/csv=true -i peacock_run_exe_tmp_test.i
Working directory: /home/agniksr/projects/leo/Input Files

*** Warning, This code is deprecated and will be removed in future versions:
Please update your main.C to adapt new main function in MOOSE framework, see'test/src/main.C in MOOSE as an example of moose::main()'.
Stack frames: 6
0: libMesh::print_trace(std::ostream&)
1: void moose::internal::mooseDeprecatedStream<libMesh::BasicOStreamProxy<char, std::char_traits<char> >, char const (&) [136]>(libMesh::BasicOStreamProxy<char, std::char_traits<char> >&, bool, bool, char const (&) [136])
2: AppFactory::createAppShared(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, int, char**, int)
3: main
4: __libc_start_main
5: /home/agniksr/projects/leo/leo-opt(+0x33ef) [0x563654a323ef]

*** ERROR ***
/home/agniksr/projects/leo/Input Files/peacock_run_exe_tmp_test.i:7.5: no variable 'core_outer_radius' found for substitution expression
/home/agniksr/projects/leo/Input Files/peacock_run_exe_tmp_test.i:7.5: no variable 'rpv_core_gap_size' found for substitution expression
/home/agniksr/projects/leo/Input Files/peacock_run_exe_tmp_test.i:7.5: no variable 'rpv_width' found for substitution expression
/home/agniksr/projects/leo/Input Files/peacock_run_exe_tmp_test.i:16.5: no variable 'core_outer_radius' found for substitution expression
/home/agniksr/projects/leo/Input Files/peacock_run_exe_tmp_test.i:30.5: no variable 'rpv_inner_radius' found for substitution expression
/home/agniksr/projects/leo/Input Files/peacock_run_exe_tmp_test.i:31.5: no variable 'rpv_outer_radius' found for substitution expression
/home/agniksr/projects/leo/Input Files/peacock_run_exe_tmp_test.i:63.5: no variable 'core_blocks' found for substitution expression
/home/agniksr/projects/leo/Input Files/peacock_run_exe_tmp_test.i:78.5: no variable 'rpv_outer_htc' found for substitution expression
/home/agniksr/projects/leo/Input Files/peacock_run_exe_tmp_test.i:79.5: no variable 'rpv_outer_Tinf' found for substitution expression
/home/agniksr/projects/leo/Input Files/peacock_run_exe_tmp_test.i:100.5: no variable 'core_blocks' found for substitution expression
/home/agniksr/projects/leo/Input Files/peacock_run_exe_tmp_test.i:115.5: no variable 'core_blocks' found for substitution expression
/home/agniksr/projects/leo/Input Files/peacock_run_exe_tmp_test.i:121.5: no variable 'core_blocks' found for substitution expression
/home/agniksr/projects/leo/Input Files/peacock_run_exe_tmp_test.i:127.5: no variable 'core_blocks' found for substitution expression
/home/agniksr/projects/leo/Input Files/peacock_run_exe_tmp_test.i:132.5: no variable 'rpv_blocks' found for substitution expression
/home/agniksr/projects/leo/Input Files/peacock_run_exe_tmp_test.i:138.5: no variable 'rpv_blocks' found for substitution expression
/home/agniksr/projects/leo/Input Files/peacock_run_exe_tmp_test.i:144.5: no variable 'rpv_blocks' found for substitution expression
/home/agniksr/projects/leo/Input Files/peacock_run_exe_tmp_test.i:149.5: no variable 'core_blocks' found for substitution expression
/home/agniksr/projects/leo/Input Files/peacock_run_exe_tmp_test.i:155.5: no variable 'rpv_outer_Tinf' found for substitution expression
/home/agniksr/projects/leo/Input Files/peacock_run_exe_tmp_test.i:156.5: no variable 'rpv_outer_htc' found for substitution expression


Stack frames: 9
0: libMesh::print_trace(std::ostream&)
1: moose::internal::mooseErrorRaw(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >)
2: void mooseError<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&>(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&)
3: Moose::Builder::build()
4: MooseApp::setupOptions()
5: MooseApp::run()
6: main
7: __libc_start_main
8: /home/agniksr/projects/leo/leo-opt(+0x33ef) [0x563654a323ef]
application called MPI_Abort(MPI_COMM_WORLD, 1) - process 0
[unset]: write_line error; fd=-1 buf=:cmd=abort exitcode=1
:
system msg for write_line failure : Bad file descriptor
Finished: Exit code: 1
'''

**Reply by GiudGiud** [reply]
please use backquotes to format your posts, not straight quotes

it looks like your app is outdated. Is this a new app? Or did you just start from an old version of moose

**Reply by agniksr** [reply]
I think i updated moose recently as well as ran the make clobberall, make and run_tests for the app as well. How can i check if im using the latest update or not

**Reply by agniksr** [reply]
Hey so i updated moose again and ran all the commands as above. The code seems to work when i run it using the  opt thing but gives the same error when i use peacock.
Any idea why that happens?

**Reply by GiudGiud** [reply]
@milljm have you seen this before?

peacock is not actively maintained.
